sdk/modelsrepository/Azure.Iot.ModelsRepository/src/DtmiConventions.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Globalization;
+using System.Text.RegularExpressions;
+
+namespace Azure.Iot.ModelsRepository
+{
+    internal class DtmiConventions
+    {
+        public static bool IsDtmi(string dtmi) => !string.IsNullOrEmpty(dtmi) && new Regex(@"^dtmi:[A-Za-z](?:[A-Za-z0-9_]*[A-Za-z0-9])?(?::[A-Za-z](?:[A-Za-z0-9_]*[A-Za-z0-9])?)*;[1-9][0-9]{0,8}$").IsMatch(dtmi);
+        public static string DtmiToPath(string dtmi) => IsDtmi(dtmi) ? $"{dtmi.ToLowerInvariant().Replace(":", "/").Replace(";", "-")}.json" : null;
+
+        public static string DtmiToQualifiedPath(string dtmi, string basePath, bool fromExpanded = false)
+        {
+            string dtmiPath = DtmiToPath(dtmi);
+            if (dtmiPath == null)
```

Comment:
Another task, to always use curly braces

Reply:
task added